code/modules/scripting/Implementations/_Logic.dm

```diff
@@ -1,144 +1,3 @@
-				if(istype(e, /list))
-					chosenlist = e
-			i = 2
-		else
-			if(chosenlist)
-				chosenlist.Add(e)
-
-// Clone of list.Remove()
-/proc/n_listremove()
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/n_listremove() called tick#: [world.time]")
-	var/list/chosenlist
-	var/i = 1
-	for(var/e in args)
-		if(i == 1)
-			if(isobject(e))
-				if(istype(e, /list))
-					chosenlist = e
-			i = 2
-		else
-			if(chosenlist)
-				chosenlist.Remove(e)
-
-// Clone of list.len = 0
-/proc/n_listcut(var/list/L, var/start, var/end)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/n_listcut() called tick#: [world.time]")
-	if(!istype(L, /list)) return
-	return L.Cut(start, end)
-
-// Clone of list.Swap()
-/proc/n_listswap(var/list/L, var/firstindex, var/secondindex)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/n_listswap() called tick#: [world.time]")
-	if(!istype(L, /list)) return
-	if(L.len >= secondindex && L.len >= firstindex)
-		return L.Swap(firstindex, secondindex)
-
-// Clone of list.Insert()
-/proc/n_listinsert(var/list/L, var/index, var/element)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/n_listinsert() called tick#: [world.time]")
-	if(!istype(L, /list)) return
-	return L.Insert(index, element)
-
-// --- Miscellaneous functions ---
-
-// Clone of rand()
-/proc/rand_chance(var/low = 0, var/high)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/rand_chance() called tick#: [world.time]")
-	return rand(low, high)
-
-// Clone of prob()
-/proc/prob_chance(var/chance)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/prob_chance() called tick#: [world.time]")
-	return prob(chance)
-
-// Merge of list.Find() and findtext()
-/proc/smartfind(var/haystack, var/needle, var/start = 1, var/end = 0)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/smartfind() called tick#: [world.time]")
-	if(haystack && needle)
-		if(isobject(haystack))
-			if(istype(haystack, /list))
-				if(length(haystack) >= end && start > 0)
-					var/list/listhaystack = haystack
-					return listhaystack.Find(needle, start, end)
-
-		else
-			if(istext(haystack))
-				if(length(haystack) >= end && start > 0)
-					return findtext(haystack, needle, start, end)
-
-// Clone of copytext()
-/proc/docopytext(var/string, var/start = 1, var/end = 0)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/docopytext() called tick#: [world.time]")
-	if(istext(string) && isnum(start) && isnum(end))
-		if(start > 0)
-			return copytext(string, start, end)
-
-// Clone of length()
-/proc/smartlength(var/container)
-	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/smartlength() called tick#: [world.time]")
-	if(container)
-		if(istype(container, /list) || istext(container))
-			return length(container)
-	return 0
-

```

Comment:
Or either of the two below.

Reply:
n_lower is only used once here:
https://github.com/ParadiseSS13/Paradise/blob/1fad05771dbb6e1ffc8eda897ac6d30b7f089f84/code/modules/admin/admin_verbs.dm#L728

n_upper isn't used anywhere, but since it's related to n_lower I wasn't sure what to do with it...

Reply:
Wait, n_lower is literally just a call to lowertext, I could just replace n_lower here with lowertext?

Reply:
Okay, removed. But github still thinks this file is a thing for some reason